### PR TITLE
feat: add blockNum to median and median deviation queries

### PR DESCRIPTION
**Pull request by zarazan**
## Description

closes: #XXXX

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/oracle/keeper/historic_price_test.go [hunk 1]
```diff
@@ -35,13 +36,15 @@ func (s *IntegrationTestSuite) TestSetHistoraclePricing() {
...
-	s.Require().Equal(medians[1], sdk.MustNewDecFromStr("1.125"))
-	s.Require().Equal(medians[2], sdk.MustNewDecFromStr("1.1"))
+	s.Require().Equal(medians[0], *types.NewPrice(sdk.MustNewDecFromStr("1.2"), displayDenom, 17))
+	s.Require().Equal(medians[1], *types.NewPrice(sdk.MustNewDecFromStr("1.125"), displayDenom, 14))
+	s.Require().Equal(medians[2], *types.NewPrice(sdk.MustNewDecFromStr("1.1"), displayDenom, 11))
 
 	medianDeviation, err := app.OracleKeeper.HistoricMedianDeviation(ctx, displayDenom)
 	s.Require().NoError(err)
-	s.Require().Equal(medianDeviation, sdk.MustNewDecFromStr("0.111803398874989476"))
+	//s.Require().Equal(medianDeviation, []types.Price{*types.NewPrice(sdk.MustNewDecFromStr("0.111803398874989476"), displayDenom, 1)})
```
Comment (rbajollari) [inline x/oracle/keeper/historic_price_test.go:45]: Should this comment still be here?

File: x/oracle/abci_test.go [hunk 2]
```diff
@@ -227,11 +227,12 @@ func (s *IntegrationTestSuite) TestEndblockerHistoracle() {
...
 			medians := app.OracleKeeper.HistoricMedians(ctx, denom.SymbolDenom, 6)
 			s.Require().Equal(4, len(medians))
-			s.Require().Equal(tc.expectedHistoricMedians, medians)
+
+			s.Require().Equal(tc.expectedHistoricMedians, medians.Decs())
 
 			medianHistoricDeviation, err := app.OracleKeeper.HistoricMedianDeviation(ctx, denom.SymbolDenom)
 			s.Require().NoError(err)
-			s.Require().Equal(tc.expectedHistoricMedianDeviation, medianHistoricDeviation)
+			s.Require().Equal(tc.expectedHistoricMedianDeviation, medianHistoricDeviation.ExchangeRateTuple.ExchangeRate)
```
Comment (rbajollari) [inline x/oracle/abci_test.go:235]: We might want to check that the block num set in these medians is as expected in this test (not just the price), but that could be in another PR down the road.

**Review by rbajollari — APPROVED** [review]
LGTM!

File: x/oracle/abci_test.go (hunk 2 above)
Comment (zarazan) [reply x/oracle/abci_test.go:235]: I agree, we could do that when this file gets cleaned up and some of this logic is moved elsewhere.

File: x/oracle/keeper/historic_price_test.go (hunk 1 above)
Comment (zarazan) [reply x/oracle/keeper/historic_price_test.go]: https://github.com/umee-network/umee/pull/1717/commits/79f914e3d9615ba81ce4cee9ada3e5841724a56e
